ogr/ogrsf_frmts/miramon/mm_gdal_functions.c

```diff
@@ -0,0 +1,2801 @@
+}  // End of MM_SprintfDoubleWidth()
+
+static MM_BOOLEAN MM_EmptyString_function(const char *cadena)
+{
+    char *ptr;
+
+    for (ptr = (char *)cadena; *ptr; ptr++)
+        if (*ptr != ' ' && *ptr != '\t')
+            return FALSE;
+
+    return TRUE;
+}
+
+int MM_SecureCopyStringFieldValue(char **pszStringDst, const char *pszStringSrc,
+                                  MM_EXT_DBF_N_FIELDS *nStringCurrentLenght)
+{
+    if (!pszStringSrc)
+    {
+        if (1 >= *nStringCurrentLenght)
+        {
+            (*pszStringDst) = realloc_function(*pszStringDst, 2);
+            if (!(*pszStringDst))
+                return 1;
+            *nStringCurrentLenght = (MM_EXT_DBF_N_FIELDS)2;
+        }
+        strcpy(*pszStringDst, "\0");
+        return 0;
+    }
+
+    if (strlen(pszStringSrc) >= *nStringCurrentLenght)
+    {
+        (*pszStringDst) =
+            realloc_function(*pszStringDst, strlen(pszStringSrc) + 1);
+        if (!(*pszStringDst))
+            return 1;
+        *nStringCurrentLenght = (MM_EXT_DBF_N_FIELDS)(strlen(pszStringSrc) + 1);
+    }
+    strcpy(*pszStringDst, pszStringSrc);
+    return 0;
+}
+
+// This function assumes that all the file is saved in disk and closed.
+int MM_ChangeDBFWidthField(struct MM_DATA_BASE_XP *data_base_XP,
+                           MM_EXT_DBF_N_FIELDS nIField,
+                           MM_BYTES_PER_FIELD_TYPE_DBF nNewWidth,
+                           MM_BYTE nNewPrecision,
+                           MM_BYTE que_fer_amb_reformatat_decimals)
+{
+    char *record, *whites = nullptr;
+    MM_BYTES_PER_FIELD_TYPE_DBF l_glop1, l_glop2, i_glop2;
+    MM_EXT_DBF_N_RECORDS nfitx, i_reg;
+    int canvi_amplada;  // change width
+    GInt32 j;
+    MM_EXT_DBF_N_FIELDS i_camp;
+    size_t retorn_fwrite;
+    int retorn_TruncaFitxer;
+
+    MM_BOOLEAN error_sprintf_n_decimals = FALSE;
+
+    canvi_amplada = nNewWidth - data_base_XP->pField[nIField].BytesPerField;
+
+    if (data_base_XP->nRecords != 0)
+    {
+        l_glop1 = data_base_XP->pField[nIField].AcumulatedBytes;
+        i_glop2 = l_glop1 + data_base_XP->pField[nIField].BytesPerField;
+        if (nIField == data_base_XP->nFields - 1)
+            l_glop2 = 0;
+        else
+            l_glop2 = data_base_XP->BytesPerRecord -
+                      data_base_XP->pField[nIField + 1].AcumulatedBytes;
+
+        if ((record = calloc_function((size_t)data_base_XP->BytesPerRecord)) ==
+            nullptr)
+            return 1;
+
+        record[data_base_XP->BytesPerRecord - 1] = MM_SetEndOfString;
+
+        if ((whites = (char *)calloc_function((size_t)nNewWidth)) == nullptr)
+        {
+            free_function(record);
+            return 1;
+        }
+        memset(whites, ' ', nNewWidth);
+
+        nfitx = data_base_XP->nRecords;
+
+#ifdef _MSC_VER
+#pragma warning(disable : 4127)
+#endif
+        for (i_reg = (canvi_amplada < 0 ? 0 : nfitx - 1); TRUE;)
+#ifdef _MSC_VER
+#pragma warning(default : 4127)
+#endif
```

Comment:
instead of shutting down warning, it would be probably clearer to rewrite as
```suggestion
        i_reg = (canvi_amplada < 0 ? 0 : nfitx - 1);
        while (TRUE)
```

Reply:
Done